icechunk/src/change_set.rs

```diff
@@ -75,45 +75,39 @@ impl ChangeSet {
         self.new_arrays.get(path)
     }
 
+    /// IMPORTANT: This method does not delete children. The caller
```

Comment:
I think this is the cleanest solution. `Session.delete_group` already builds a list of nodes to delete. We could filter those to not be children of any groups added in this session (and may some other similarly complex predicate), or we can just not delete children ...